Topics/Development_Process/Shell/intro_to_shell.md

```diff
@@ -0,0 +1,159 @@
+
+- `mkdir`: Creates a new directory.
+  - **Usage:** `mkdir [directory name]`
+  - **Example:** `mkdir new_folder` - Creates a new directory named `new_folder`.
+
+## Copying and Moving Files
+
+- `cp`: Copies files or directories.
+  - **Usage:** `cp [source] [destination]`
+  - **Options:** `-r` for recursive copying (directories), `-i` for interactive mode.
+  - **Example:** `cp file.txt ~/Documents/` - Copies `file.txt` to the Documents directory.
+
+- `mv`: Moves or renames files or directories.
+  - **Usage:** `mv [source] [destination]`
+  - **Example:** `mv oldname.txt newname.txt` - Renames `oldname.txt` to `newname.txt`.
+
+## Deleting Files and Directories
+
+- `rm`: Removes files or directories.
+  - **Usage:** `rm [file or directory]`
+  - **Options:** `-r` for recursive deletion (directories), `-f` for force deletion without prompt.
+  - **Example:** `rm -rf old_folder/` - Forcefully removes `old_folder` and its contents.
+
+## Tips for Safe File Manipulation
+
+1. **Double-check file names and paths before executing commands**, especially with `rm`.
+2. **Use wildcards (`*`) carefully**, as they can affect multiple files.
+3. **Regularly back up important files** to avoid accidental loss.
+
+These commands form the foundation of file management in the Unix-like systems and are essential for day-to-day operations.
+
+
+# Viewing and Editing Files
+
+Being able to view and edit files directly from the command line is a valuable skill. Here's how to do it:
+
+## Viewing File Contents
+
+- `cat`: Concatenates and displays the content of files.
+  - **Usage:** `cat [file]`
+  - **Example:** `cat notes.txt` - Displays the content of `notes.txt`.
+
+- `less`: Allows for perusal of text files one screen at a time.
+  - **Usage:** `less [file]`
+  - **Example:** `less log.txt` - Views `log.txt` in a scrollable format.
+
+## Editing Files
+
+- `nano`: A simple, easy-to-use text editor.
+  - **Usage:** `nano [file]`
+  - **Example:** `nano diary.txt` - Opens `diary.txt` for editing in nano.
+
+- `vi`/`vim`: More advanced text editors, offering a wide range of features.
+  - **Usage:** `vi [file]`
+  - **Example:** `vi script.sh` - Opens `script.sh` for editing in vi.
+
+## Tips for Editing Files in the CLI
+
+1. **Learn basic navigation shortcuts**, like moving with arrow keys, page up/down in `less`.
+2. **Familiarize yourself with text editor commands**, especially if using `vi` or `vim`.
+3. **Regularly save your work** to avoid data loss.
+
+Mastering these viewing and editing commands is essential for efficient software development and system administration.
+
+# Best Practices
+
+Adhering to best practices in the command line ensures efficiency, safety, and reliability. Here are some key practices to follow:
+
+1. **Command Line Proficiency:** Regularly practice common commands and explore new ones.
+2. **Scripting:** Automate repetitive tasks with shell scripts to save time.
```

Comment:
For writing shell scripts, I think this is another topic, and you can probably just link another reference to writing shell scripts in this file. Better to not leave the readers hanging because this markdown currently doesn't tell you anything about shell scripts.

Reply:
O I checked down there, but maybe you can make a reference that claims something like: 
"You can learn more by checking the additional resources in the following section"

Reply:
Yes, I have mentioned to look for additional resources in the following section. But still thank you for checking this.